compiler/damlc/tests/daml-test-files/RecordUpdate.daml

```diff
@@ -61,10 +61,12 @@ data Point = Point with
 origin = Point with x = 0; y = 0
 
 -- NOTE(MH): We expect this to become
--- `ERecUpd _ "x" (EVal ["origin"]) (EBuiltin (BEInt64 1))`
+-- NOTE(drsk): The ELet binding was added with RecordDotUpdates for better type inference.
+-- `ELet (Binding ("arg", _) (EVal ["origin"])) (ERecUpd _ "x" (EVal ["arg"]) (EBuiltin (BEInt64 1)))`
```

Comment:
I’m slightly worried about this. The `let` for non-nested updates does(’t help with inference. It just slows things down. Can we easily detect a singleton field list in GHC and not generate the case there?

Reply:
I can think of two things: Either traverse the updates and check whether we have nested ones or move the Let/Case bindings to he update expression instead outside of the record update and only add it for nested updates. But that would potentially mean many let/case bindings.

Reply:
This has been changed now in the ghc fork to only add the case/let expression for nested updates.